.github/workflows/publish.yml

```diff
@@ -0,0 +1,106 @@
+
+name: Publish Artifacts
+
+on:
+  push:
+    branches: [main]
```

Comment:
This doesn't seem right.
Looks like its going to run this on every merge to main?

Reply:
Perhaps, a release branch would make more sense?

Reply:
It seems like the intent would be to run this on tag rather than on a branch.

Reply:
The intention is to run the `publish` action after the version bump PR merged to `main` branch